### ISSUE TITLE
file-annotation in Kotlin DSL pre-compiled script plugin with package disturbs discovery

### ISSUE THREAD

**Issue by Vampire** [gradle/gradle#16154]
Given in Gradle 6.8.2:
`build.gradle.kts`:
```kts
plugins {
    id("bar.foo")
}
```
`settings.gradle.kts`:
```kts
includeBuild("build-src")
```
`build-src/build.gradle.kts`:
```kts
plugins {
    `kotlin-dsl`
}

repositories {
    mavenCentral()
}
```
`build-src/src/main/kotlin/foo.gradle.kts`:
```kts
package bar

plugins {
    base
}
```

This constellation works perfectly fine.
But if you now change `build-src/src/main/kotlin/foo.gradle.kts` to:
```kts
@file:Suppress("UnstableApiUsage")

package bar

plugins {
    base
}
```

the build breaks with `Plugin [id: 'bar.foo'] was not found in any of the following sources`.
Also changing to `id("foo")` does not make it better, as then it finds the right plugin to apply but fails as the class `Foo_gradle` is not found (there is only `bar.Foo_gradle`).

Currently the only way to have a file-level annotation is by not using a package statement, but put the package into the class name.
So if you remove the `package bar` and rename the file to `bar.foo.gradle.kts`, you can again use `id("bar.foo")`.

**Comment by cesar1000** [comment]
I just ran into the same issue, and it looks like `@file:Suppress("UnstableApiUsage")` (which is not that unusual anyway) causes script parsing to fail silently. I put up a trivial [repro](https://github.com/cesar1000/package-parsing) that shows the extracted plugin block isn't right:

```
$ ./gradlew :build

BUILD SUCCESSFUL in 3s
13 actionable tasks: 12 executed, 1 up-to-date

$ ls /Users/cpuerta/temp/gradle_issue_16154_package_parsing/build/generated-sources/kotlin-dsl-plugins/kotlin/
MyPluginWithSuppressPlugin.kt  com/

$ cat build/kotlin-dsl/plugins-blocks/extracted/my-plugin-with-suppress.gradle.kts




plugins {
    java
}

$ cat build/kotlin-dsl/plugins-blocks/extracted/com/twitter/my-plugin.gradle.kts
package com.twitter;

plugins {
    java
}
```

**Comment by akuleshov7** [comment]
You can use this reproducer: https://github.com/akuleshov7/gradle-issue-repro, but I am pretty sure that the problem is obvious

**Comment by jbartok** [comment]
Fixed by 
* #25131

**Comment by Vampire** [comment]
I think you should report this as a new regression issue @elect86.
The actual error here was indeed fixed in 8.3 and is covered by tests.
The current error is a new one, that was introduced in 8.7.
In 8.6 it still works fine.
